tools/spec2vec/spec2vec_training.xml

```diff
@@ -0,0 +1,254 @@
+                    <option value="0">Negative Sampling</option>
+                    <option value="1">Hierarchical Softmax</option>
+                </param>
+                <when value="0">
+                    <param label="Negative Samples" name="negative" type="integer" value="5"
+                        help="Specify how many 'negative' examples should be drawn for each peak and neutral loss (usually between 5-20).">
+                        <validator type="in_range" min="1" message="The value must be larger than 0."/>
+                    </param>
+                    <param label="Negative Sample Exponent" name="ns_exponent" type="float" value="0.75"
+                        help="The exponent used to shape the negative sampling distribution. A value of 1.0 samples exactly in proportion to the frequencies,
+                        0.0 samples all peaks and neutral losses equally, while a negative value samples low-frequency peaks more often than high-requency peaks.">
+                        <validator type="in_range" min="-1.0" max="1.0" message="The value must be within -1.0 and 1.0 range."/>
+                    </param>
+                </when>
+            </conditional>
+            <param label="Sort the vocabulary of spectra" name="sorted_vocab" type="boolean" checked="true" truevalue="TRUE" falsevalue="FALSE"
+                   help="If true, sort the vocabulary by descending frequency before assigning peak and neutral loss indices."/>
+            <param label="Batch size" name="batch_words" type="integer" value="10000"
+                   help="Target size (in peaks and neutral losses) for batches of examples passed to worker threads (and thus cython routines).
+                   Larger batches will be passed if individual peak sequences are longer than 10000 words, but the standard cython code truncates to that maximum."/>
+            <param label="Shrink windows" name="shrink_windows" type="boolean" checked="true" truevalue="TRUE" falsevalue="FALSE"
+                   help="EXPERIMENTAL. If true, the effective window size is uniformly sampled in range [1,Window] for each target peak during training."/>
+            <conditional name="trim_vocab">
+                <param label="Maximum unique peaks and neutral losses in the spectral vocabulary" name="max_vocab_size_bool" type="boolean" truevalue="TRUE" falsevalue="FALSE" checked="false"
+                        help="Limits the RAM during vocabulary building; if there are more unique peaks and neutral losses than this, then prune the infrequent ones. Disable for no limit (default)."/>
+                <when value="TRUE">
+                    <param label="Maximum unique peaks and neutral losses" name="max_vocab_size" type="integer" value="100000"/>
+                </when>
+            </conditional>
+            <param label="Number of decimals to round m/z values" name="n_decimals" type="integer" value="2"
```

Comment:
You could add min/max to those paramters.

Reply:
That is indeed a good idea!

Reply:
Added more validators in 662d691.